x/incentive/types/codec.go

```diff
@@ -0,0 +1,18 @@
+package types
+
+import "github.com/cosmos/cosmos-sdk/codec"
+
+// ModuleCdc generic sealed codec to be used throughout module
+var ModuleCdc *codec.Codec
+
+func init() {
+	cdc := codec.New()
+	RegisterCodec(cdc)
+	codec.RegisterCrypto(cdc)
+	ModuleCdc = cdc.Seal()
+}
+
+// RegisterCodec registers the necessary types for incentive module
+func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterConcrete(MsgClaimReward{}, "incentive/MsgClaimReward", nil)
```

Comment:
is this the only type that needs to be registered?

Reply:
I believe so? There is only one message type and all other types are non-interfaces. Is there a good resource for determining what types to register?

Reply:
messages, types that are stored and interfaces